src/ostorlab/agent/agent.py

```diff
@@ -395,6 +395,37 @@ def _prepare_message(self, raw: bytes) -> bytes:
         )
         return control_message.raw
 
+    def _prepare_in_selectors(
+        self,
+        agent_definition: agent_definitions.AgentDefinition,
+        agent_settings: runtime_definitions.AgentSettings,
+    ) -> List[str]:
+        """Prepare the in selectors, taking into account the extended selectors.
+
+        Args:
+            agent_definition: Agent definition dictating the attributes of the agent.
+            agent_settings: The settings of the running instance of an agent.
+
+        Returns:
+            List of in selectors.
+        """
+        in_selectors = set(agent_definition.in_selectors)
+        extended_selectors = set(agent_settings.extended_in_selectors)
+
+        in_selectors = [
+            selector
+            for selector in in_selectors
+            if any(
+                selector in extended_selector
+                for extended_selector in extended_selectors
+            )
+            is False
+        ]
+
+        in_selectors.extend(extended_selectors)
+
```

Comment:
extended selectors should simply override the selectors. For this reason I don't think `extended` is the right name in the rest of the code.

The behavior we want to achieve is to replace the existing ones with the provided ones.

Reply:
@3asm i will rename to `runtime_in_selectors` !